src/oneNoteDataStructures/oneNoteItemUtils.ts

```diff
@@ -98,6 +100,38 @@ export class OneNoteItemUtils {
 
 		return ancestry;
 	}
+	
+	/**
+	 * Finds the maximum depth of notebooks list, including sections
+	 */
+	getDepthOfNotebooks(notebooks: Notebook[]): number {
+		if (!notebooks || notebooks.length === 0) {
+			return 0;
+		}
+		return notebooks.map((notebook) => this.getDepthOfParent(notebook)).reduce((d1, d2) => Math.max(d1, d2));
+	}
+
+	/** 
+	 * Finds the maximum depth of SectionGroup or Notebook, 
+	 * includes the number of sections below it 
+	 */
+	getDepthOfParent(parent: noteBookOrSectionGroup ): number {
+		if (!parent) {
+			return 0;
+		}
+
+		let containsAtLeastOneSection = parent.sections && parent.sections.length > 0;
+		let maxDepth = containsAtLeastOneSection ? 1 : 0;
+
+		if (parent.sectionGroups) {
+			for (let i = 0; i < parent.sections.length; i++) {
```

Comment:
I think you meant parent.sectiongroups here. Also, I suggest using
for(const sectionGroup of parent.sectionGroups){
...
instead of a for with indexes

Reply:
I think the sectionGroups thing is actually to measure the sections here- it looks weird but changing these around breaks it. I'm mostly copying this code over from the utils in the OneNoteAPI repo we were using before

Reply:
The code says:
---
if (parent.sectionGroups) {
		for (let i = 0; i < parent.sections.length; i++) {
				maxDepth = Math.max(this.getDepthOfParent(parent.sectionGroups[i]), maxDepth);
			}
		}
---
Let's suppose this parent has 1 section groups and 10 sections. It goes inside the if, loops from 0 to 9, and tries to access parent.sectionGroups[i]; It will throw when i=1 because the parent only has 1 section group.

Reply:
Does this mean there's a bug in the OneNoteApi repo we should fix?